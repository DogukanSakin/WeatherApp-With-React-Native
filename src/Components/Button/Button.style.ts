import { StyleSheet } from 'react-native';
import Fonts from '../../Styles/Fonts';
const baseButtonStyle=StyleSheet.create({
    buttonText:{
        fontFamily:Fonts.semiBoldFont,
        fontSize:20
    },
    buttonContainer:{
        justifyContent:'center',
        alignItems:'center',
        padding:10,
        marginTop:15,
        borderRadius:50
    }
})
export default {
    primary:StyleSheet.create({

        buttonText:{
            ...baseButtonStyle.buttonText,
            color:'white'
        },
        buttonContainer:{
            ...baseButtonStyle.buttonContainer,
            backgroundColor:'#5BA7FB'
        }
    }),
    secondary:StyleSheet.create({
        buttonText:{
            ...baseButtonStyle.buttonText,
            color:'#5BA7FB'
        },
        buttonContainer:{
            ...baseButtonStyle.buttonContainer,
            borderWidth:1,
            borderColor:'#5BA7FB'
        }

    })

}