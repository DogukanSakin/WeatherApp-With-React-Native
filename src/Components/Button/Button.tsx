import React,{FC} from 'react';
import { Text,TouchableOpacity, TouchableOpacityProps } from 'react-native';
import styles from './Button.style';
interface IProps extends TouchableOpacityProps{
    buttonTitle:string;
    theme?:'primary' | 'secondary';
}
const Button:FC<IProps>=({buttonTitle,theme='primary',...rest})=>{
    return(
        <TouchableOpacity style={styles[theme].buttonContainer} {...rest}>
            <Text style={styles[theme].buttonText}>{buttonTitle}</Text>
        </TouchableOpacity>
    )
}
export default Button;