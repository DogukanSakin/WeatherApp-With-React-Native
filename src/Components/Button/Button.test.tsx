import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import Button from './Button';
import styles from './Button.style';
test('should match with snapshot', () => { 
    const comp=render(<Button buttonTitle='Test'/>);
    expect(comp).toMatchSnapshot();
})
test('should  render title correctly', () => { 
    const testTitle='This is test title!';
    const comp=render(<Button buttonTitle={testTitle}/>);
    const buttonText= comp.getByTestId('button-title').children[0];
    expect(buttonText).toBe(testTitle);
})
test('should trigger onPress', () => { 
    const mockFunction=jest.fn();
    const comp=render(<Button buttonTitle='Test' onPress={mockFunction}/>);
    const buttonTocuhable=comp.getByTestId('touchable-button');
    fireEvent(buttonTocuhable,'press');
    expect(mockFunction).toBeCalledTimes(1);
})
test('should render given default theme if theme property is exist', () => { 
    const comp=render(<Button buttonTitle='Test'/>);
    const button=comp.getByTestId('touchable-button').props.style;
    expect(button).toMatchObject(styles.primary.buttonContainer);
})
test('should render given secondary theme', () => { 
    const style="secondary";
    const comp=render(<Button buttonTitle='Test' theme={style}/>);
    const button=comp.getByTestId('touchable-button').props.style;
    expect(button).toMatchObject(styles[style].buttonContainer);
})
