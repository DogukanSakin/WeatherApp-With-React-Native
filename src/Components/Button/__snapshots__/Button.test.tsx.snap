// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#5BA7FB",
      "borderRadius": 50,
      "justifyContent": "center",
      "marginTop": 15,
      "opacity": 1,
      "padding": 10,
    }
  }
  testID="touchable-button"
>
  <Text
    style={
      Object {
        "color": "white",
        "fontFamily": "OpenSans-SemiBold",
        "fontSize": 20,
      }
    }
    testID="button-title"
  >
    Test
  </Text>
</View>
`;
